src/test/java/com/cloudbees/hudson/plugins/folder/FolderTest.java

```diff
@@ -225,6 +233,27 @@ private void copyViaHttp(Folder f, JenkinsRule.WebClient wc, String fromName, St
         }
     }
 
+    @Issue("JENKINS-35160")
+    @Test
+    public void interruptOnDelete() throws Exception {
+        // adapted from JobTest in core
+        r.jenkins.setNumExecutors(2);
+        Queue.getInstance().maintain();
+        Folder d = r.createProject(Folder.class, "d");
+        FreeStyleProject p = d.createProject(FreeStyleProject.class, "p");
+        p.addProperty(new ParametersDefinitionProperty(new StringParameterDefinition("dummy", "0")));
+        p.setConcurrentBuild(true);
+        p.getBuildersList().add(new SleepBuilder(Long.MAX_VALUE));
+        FreeStyleBuild build1 = p.scheduleBuild2(0).getStartCondition().get();
+        FreeStyleBuild build2 = p.scheduleBuild2(0).getStartCondition().get();
+        QueueTaskFuture<FreeStyleBuild> build3 = p.scheduleBuild2(0);
+        Thread.sleep(1000); // TODO Queue.cancel(Item) can return false immediately after scheduling
+        d.delete();
+        r.assertBuildStatus(Result.ABORTED, r.waitForCompletion(build1));
+        r.assertBuildStatus(Result.ABORTED, r.waitForCompletion(build2));
+        assertTrue(build3.isCancelled());
```

Comment:
Actually this passes even with `src/main/` from this PR and `pom.xml` from `master`, i.e. using the original inherited `AbstractItem.delete` implementation which already cancelled builds and queue items for folder children. It would not call `FreeStyleProject.delete` prior to `Folder.delete` but there was not much practical impact (perhaps only missing calls to `ItemListener.fireOnDeleted`) since all of `$JENKINS_HOME/jobs/d/` was deleted.